src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -385,27 +385,28 @@ public int CompareTo(object? obj)
         /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
         public int CompareTo(Half other)
         {
-            if (this < other)
-            {
-                return -1;
-            }
+            bool thisIsNaN = IsNaN(this);
+            bool otherIsNaN = IsNaN(other);
 
-            if (this > other)
-            {
-                return 1;
-            }
+            if (thisIsNaN || otherIsNaN)
```

Comment:
With the way the JIT emits branches by default, this is probably less ideal because it will assume one of the inputs is NaN, while NaN is typically an edge case in practice.

It might be beneficial to invert this to be "if neither are nan"